Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -9,20 +13,18 @@
 
 namespace Snowflake.Data.Core.Session
 {
-    sealed class SessionPoolSingleton : IDisposable
+    sealed class SessionPool : IDisposable
```

Comment:
Maybe call it SessionCache?

Reply:
this name was for long here and it is actually a mechanism to pool sessions so why to change?